WordPress/gutenberg
# Mobile - RichText - Fix issue with font sizes with decimal values

- Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4324

## Description
While testing the upcoming Twenty Twenty-Two theme, I encountered a crash while opening the editor on the app.

After debugging I saw the font sizes values from the `theme.json` had decimals that weren't properly parsed by `getPxFromCssUnit`.

Font sizes from the [theme.json](https://github.com/WordPress/twentytwentytwo/blob/trunk/theme.json):

```json
{
    "fontSizes": [
        {
            "name": "Small",
            "size": "1rem",
            "slug": "small"
        },
        {
            "name": "Normal",
            "size": "1.125rem",
            "slug": "normal"
        },
        {
            "name": "Medium",
            "size": "1.75rem",
            "slug": "medium"
        },
        {
            "name": "Large",
            "size": "clamp(1.75rem, 3vw, 2.25rem)",
            "slug": "large"
        },
        {
            "name": "Huge",
            "size": "clamp(2.5rem, 4vw, 3rem)",
            "slug": "huge"
        }
    ]
}
```

This PR modifies the Regex expression to match these values as well. I also added a fallback for the font size in cases where `getPxFromCssUnit` returns an undefined value.

## How has this been tested?

**Precondition**: A test site with the Twenty Twenty-Two theme, you can download it [here](https://github.com/WordPress/twentytwentytwo/).

- Open the editor
- **Expect** to see the editor without crashing
- Open the block settings of a Paragraph block
- Open the font size picker
- **Expect** to see the list of font sizes

## Screenshots 

Editor with the TT2 Theme | Editor with the TT2 Theme (Font size picker)
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/144238087-7646b409-14e2-424b-b776-774b130864c4.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/144238111-f3b20dd3-e103-4fe4-b0e9-83a25223ac74.png" width="200" /></kbd>

## Types of changes
Bug fix

## Checklist:
